tools/ARIAtools/extractProduct.py

```diff
@@ -664,59 +666,57 @@ def export_products(full_product_dict, bbox_file, prods_TOTbbox, layers,
 
             # make VRTs for wet and then dry components
-            hgt_field = prep_metadatalayers(wt_outname, [i[1]][0], dem)
-            hy_outname = os.path.abspath(os.path.join(workdir, end_key
-                                         + product_dict[1][i[0]][0]))
             tropo_prefix = '/science/grids/corrections/external/troposphere/'
-            hydro_comp = [i.replace(tropo_prefix+start_key, \
-                          tropo_prefix+end_key) \
-                          for i in [i[1]][0]]
-            hgt_field = prep_metadatalayers(hy_outname, hydro_comp, dem)
-
-
-            # initiate raster
-            outname = os.path.abspath(os.path.join(workdir,
-                                      product_dict[1][i[0]][0]))
-            # building the VRT
-            gdal.BuildVRT(outname + '.vrt', wt_outname + '.vrt')
-            gdal.Open(outname + '.vrt').SetMetadataItem(
-                      hgt_field, gdal.Open(wt_outname + '.vrt'
-                                            ).GetMetadataItem(hgt_field))
-            gdal.Warp(outname, outname + '.vrt',
-                      options=gdal.WarpOptions(format=outputFormat))
-            # Update VRT
-            gdal.Translate(outname + '.vrt', outname,
-                           options=gdal.TranslateOptions(format="VRT"))
-            # Update VRT with new raster
-            update_file = gdal.Open(outname, gdal.GA_Update)
-            heightsMeta = np.array(gdal.Open(wt_outname + '.vrt' \
-                          ).GetMetadataItem(hgt_field)[1:-1].split(','), \
-                          dtype='float32')
-            wet_arr = gdal.Open(wt_outname + '.vrt').ReadAsArray()
-            hy_arr = gdal.Open(hy_outname + '.vrt').ReadAsArray()
-            for i in range(len(heightsMeta)):
-                tot_arr = wet_arr[i] + hy_arr[i]
-                update_file.GetRasterBand(i+1).WriteArray(tot_arr)
-            del update_file
-            # Interpolate/intersect with DEM before cropping
-            finalize_metadata(outname, bounds, dem_bounds, \
-                              prods_TOTbbox, dem, lat, lon, hgt_field, \
-                              mask, outputFormat, verbose=verbose)
+            hgt_field = prep_metadatalayers(wt_outname, [i[1]][0], dem)
+            for key in [start_key, end_key]:
+                t_outname  = os.path.abspath(os.path.join(workdir, key + ifg))
 
-            #If necessary, resample raster
-            if multilooking is not None:
-                resampleRaster(outname, multilooking, bounds, prods_TOTbbox,
-                               rankedResampling, outputFormat=outputFormat,
-                               num_threads=num_threads)
+                t_comp     = [path_nc_wet.replace(tropo_prefix+start_key, tropo_prefix+key)]
+
+                hgt_field = prep_metadatalayers(t_outname, t_comp, dem)
+
+                finalize_metadata(t_outname, bounds, dem_bounds, \
```

Comment:
@bbuzz31 this procedure is now interpolating the intermediate tropo layers even if the user may not want them in the end. This introduces a bit of extra overhead we could've circumvented as I'd intended to do before, where I initialized a total tropo array using temp files before then interpolating just that directly.